src/librustc_mir/hair/pattern/mod.rs

```diff
@@ -1152,7 +1156,11 @@ fn lit_to_const<'a, 'tcx>(lit: &'tcx ast::LitKind,
                 ty::TyInt(other) => Int::Signed(other),
                 ty::TyUint(UintTy::Usize) => Int::Unsigned(tcx.sess.target.usize_ty),
                 ty::TyUint(other) => Int::Unsigned(other),
-                _ => bug!(),
+                ty::TyError => {
+                    // Avoid ICE
+                    return Err(false);
+                }
+                _ => bug!("{:?}", ty.sty),
```

Comment:
Expand the `bug` description a little with something like "literal integer with bad type".